lib/health_checks/sidekiq.rb

```diff
@@ -15,18 +17,24 @@ def sidekiq(config)
   class LivenessServer
     LIVENESS_PORT = 8080
 
-    def start
+    def start(mongo_databases, redis_configs)
       Sidekiq::Logging.logger.info "Starting liveness server on #{LIVENESS_PORT}"
+
       Thread.start do
+        checks = mongo_databases.map { |db| Custom::MongoidCheck.new(db) }
+        checks += redis_configs.map{ |config| Custom::RedisCheck.new(config) }
+        checks << Custom::MemoryCheck.new
         server = TCPServer.new(LIVENESS_PORT)
         loop do
           Thread.start(server.accept) do |socket|
             begin
-              HealthChecks.memory
-              check_redis
-              check_mongo
+              threads = checks.map do |check|
+                Thread.new { check.run }
```

Comment:
Spawning threads all the time is expensive, always use a thread pool

Reply:
any suggestions what gem to use(or implement my own threadpool)? I've found concurrent-ruby, but it comes with a lot of other stuffs. Also, OkComputer parallel implementation is the same as mine, which means it makes it not good :(

Reply:
Yet again I point out OKComputer is quite the simple and yet lame gem, not everything open source is a standard to live up to :)
I can't find any other gem so my suggestion is to drop concurrency for now and just increase the liveness probe timeout accordingly

Reply:
and any suggestion how much to do it or what base to find out? Currently, for the workers is 5s and for web is the default 1s. I could make it 1s for each check, so 6s for workers and 4 for web?

Reply:
Time to do all checks + 1s so yes, 6 and 4

Reply:
so 7 and 5(2 mongo, 1 memory, 3 redis for the worker and the same for the web, but 1 redis)

Reply:
The memory check is not a network request, it's fast